### DISCUSSION TITLE
Can We Add Deadline

### DISCUSSION THREAD

**Discussion by dendisuhubdy** [community/community#4282] [Projects and Issues] [answered]
Can we add a deadline to the task list? Other than just this? So we can attached it to Projects so you guys can kill JIRA and Trello

<img width="375" alt="Screen Shot 2021-06-24 at 12 08 14" src="https://user-images.githubusercontent.com/4477337/123205879-db313800-d4e4-11eb-86f0-8584fb70e0f6.png">

**Comment by azenMatt** [comment, marked as answer]
Hi @dendisuhubdy - the new version of projects that we are working has a 'date' type custom field that could be used to communicate a deadline. More on this coming soon!

**Reply by 0xdevalias** [reply]
For those looking/wanting this, it looks like the 'Date' type that @azenMatt mentioned may be live now:

![image](https://user-images.githubusercontent.com/753891/126086205-cc7978dc-94f3-4a47-8f71-a61dce882ae6.png)

**Reply by johann-petrak** [reply]
@azenMatt since this is under "issues feedback", how would this relate to a due date/deadline for issues?
There is an issue asking to add due dates to issues which has been basically ignored for 6 years now, despite the fact that a lot of people would really need this feature which has been present e.g. in gitlab for a while now. 
See https://github.com/isaacs/github/issues/432
Of course this is only really useful if the due date allows for a little more than just "communicating" the date, e.g. alert approaching deadlines, select/order by deadline etc.

**Reply by mhnsn** [reply]
Lack of ability to give issues deadlines, and then sort issues by them is...embarrassingly bad. Combine that with a lack of ability to add weights or time estimates for issues, and it's catastrophic. Literally a fundamental concept of software engineering is balancing task urgency and importance, and one of the biggest companies in the world doesn't support either concept. If Microsoft and Github don't want to start seriously bleeding business to competitors who already have this functionality (and have had it for years), they need to get this fixed.

**Comment by 0xdevalias** [comment]
It would be cool if we could have a 'Date/Time' type as well as just the 'Date' (or maybe it could rolled into one that can optionally support having a time or not)?

There are times where we have our 'due by' deadlines set to a specific time, as that is when a particular event/etc may be happening. At the moment we just tend to capture this manually within the 'description' text of our issues, and use issue templates for this, but it would be awesome if we were able to capture this within the project field's themselves, as this would then be MUCH more visible, rather than needing to click into each individual issue to see if it has a due date set, etc.

**Comment by 0xdevalias** [comment]
It would also be cool if we could have some colouring/indicators to make it easier to distinguish between Dates in the past/overdue (eg. red), that are 'soon' (maybe yellow?), and that are in the future (could just be standard/without colour).

Currently they look like this in the table view (note that one of these dates is in the past, and one is in the future):
![image](https://user-images.githubusercontent.com/753891/126086691-e6fdcbee-64f3-4fa7-b888-477aa67dc1f6.png)

**Comment by bob27aggiustatutto** [comment]
I don't see any "date" option, can anyone help me find it?

**Reply by svintit** [reply]
@bob27aggiustatutto you need to make a custom field

**Reply by bob27aggiustatutto** [reply]
> @bob27aggiustatutto you need to make a custom field

found it. thank you!

**Comment by fguillen** [comment]
I still don't see where the Custom Fields are. (I am in a Card in a GitHub Project):
![Screenshot 2025-04-10 at 15 12 56](https://github.com/user-attachments/assets/24b1978a-6534-4a12-9084-7e8762c1ccd2)

**Reply by adonohoe** [reply]
For anyone else still unable to find the custom fields, they are set up against the Project, not within the Issues/Cards.

Go into the Project, then click the three-dots menu in the top right, and select Settings.
Then over on the left you should see a column with the option to create Custom Fields.

**Reply by fguillen** [reply]
It worked, thanks!
![Screenshot 2025-06-02 at 11 12 17](https://github.com/user-attachments/assets/33cb25b1-3aa3-41ff-8089-3274365721e7)
